src/Maker/MakeResetPassword.php

```diff
@@ -0,0 +1,277 @@
+            'Controller\\'
+        );
+
+        $requestClassNameDetails = $generator->createClassNameDetails(
+            'ResetPasswordRequest',
+            'Entity\\'
+        );
+
+        $repositoryClassNameDetails = $generator->createClassNameDetails(
+            'ResetPasswordRequestRepository',
+            'Repository\\'
+        );
+
+        $requestFormTypeClassNameDetails = $generator->createClassNameDetails(
+            'ResetPasswordRequestFormType',
+            'Form\\'
+        );
+
+        $changePasswordFormTypeClassNameDetails = $generator->createClassNameDetails(
+            'ChangePasswordFormType',
+            'Form\\'
+        );
+
+        $generator->generateController(
+            $controllerClassNameDetails->getFullName(),
+            'resetPassword/ResetPasswordController.tpl.php',
+            [
+                'user_full_class_name' => $userClassNameDetails->getFullName(),
+                'user_class_name' => $userClassNameDetails->getShortName(),
+                'request_form_type_full_class_name' => $requestFormTypeClassNameDetails->getFullName(),
+                'request_form_type_class_name' => $requestFormTypeClassNameDetails->getShortName(),
+                'reset_form_type_full_class_name' => $changePasswordFormTypeClassNameDetails->getFullName(),
+                'reset_form_type_class_name' => $changePasswordFormTypeClassNameDetails->getShortName(),
+                'password_setter' => $input->getArgument('password-setter'),
+                'success_redirect_route' => $input->getArgument('controller-reset-success-redirect'),
+                'from_email' => $input->getArgument('from-email-address'),
+                'from_email_name' => $input->getArgument('from-email-name'),
+                'email_getter' => $input->getArgument('email-getter'),
+            ]
+        );
+
+        $generator->generateClass(
+            $requestClassNameDetails->getFullName(),
+            'resetPassword/ResetPasswordRequest.tpl.php',
+            [
+                'repository_class_name' => $repositoryClassNameDetails->getFullName(),
+                'user_full_class_name' => $userClassNameDetails->getFullName(),
+            ]
+        );
+
+        $generator->generateClass(
+            $repositoryClassNameDetails->getFullName(),
+            'resetPassword/ResetPasswordRequestRepository.tpl.php',
+            [
+                'request_class_full_name' => $requestClassNameDetails->getFullName(),
+                'request_class_name' => $requestClassNameDetails->getShortName(),
+            ]
+        );
+
+        if (!$this->fileManager->fileExists($path = 'config/packages/reset_password.yaml')) {
+            throw new RuntimeCommandException(sprintf('The file "%s" does not exist. This command needs that file to accurately build the reset password config.', $path));
+        }
+
+        $manipulator = new YamlSourceManipulator($this->fileManager->getFileContents($path));
```

Comment:
This is a "funny" situation. We sorta don't want to "update" this file (and leave the old comments). We probably want to completely *override* the file contents with the new ones. Specifically, we don't want our "Replace symfonycasts.reset_password.fake_request_repository with the full..." text to be in this file when we finish.

Perhaps we could check if `request_password_repository: symfonycasts.reset_password.fake_request_repository` is the only config in that file. And:

A) If it is, use the normal `Yaml` file to completely dump new contents
B) If not for some reasons, then use the manipulator to update the file more carefully

Reply:
done,

1) if the config file doesn't exist, we assume dev is using a customized config. Remind dev to set correct repository and move on.

2) if the config exists with only 1 key, assume config was created by flex; create a new config file.

3) if config exists with more than 1 key, assume the dev modified the config before running maker. Replace repository value and preserve everything else.